### DISCUSSION TITLE
Does this only work for projects that have a package.json?

### DISCUSSION THREAD

**Discussion by matthewadams** [semantic-release/semantic-release#3465] [General]
We have a monorepo with several projects of differing types:  Node.js, Gradle Kotlin, C++, etc, all of which employ semantic versioning.  Additionally, we maintain a plain text `VERSION` file, too.  To make things easy, all projects in the whole monorepo are required to have the same version (yes, I know, that means the exact same bits could be released in two different versions).

Can this tool modify all files containing version strings, like multiple `build.gradle.kts`, `VERSION`, `package.json`, files?

**Comment by travi** [comment]
Two part answer:
* Related to your direct question https://semantic-release.gitbook.io/semantic-release/support/faq#can-i-use-semantic-release-to-publish-non-javascript-packages
* Related to monorepos: we do not officially support monorepos of any type. semantic-release expects releasables to be 1:1 with the repository. There are, however, community supported extensions to semantic-release that as monorepo support, but they are not supported by this team
